app/src/main/java/org/piwigo/ui/login/LoginViewModel.java

```diff
@@ -219,20 +218,21 @@ public void onCompleted() {
         public void onError(Throwable e) {
             Log.e(TAG, e.getMessage());
             loginError.setValue(e);
+            if (url.get() != null && url.get().contains("https"))
+                testConnection(true, URLHelper.INSTANCE.getUrlWithMethod(url.get(), "http"));
```

Comment:
I don't fully catch this code. What is the intention here?

Reply:
Retry with the "http" protocol when "https" have been already tried and avoid a loop thanks to the "contains(https)"

Reply:
Ok. I don't see why it should not work, but the control flow is not really easy to catch here. If you see how to simplify that it would be great, otherwise consider adding a few comments.